Beanbag explodes in mvnd 1.0.0

With strange error:
```
Caused by: java.lang.NoClassDefFoundError: org/jline/reader/LineReader
    at java.lang.Class.getDeclaredFields0 (Native Method)
    at java.lang.Class.privateGetDeclaredFields (Class.java:3473)
    at java.lang.Class.getDeclaredFields (Class.java:2542)
    at io.smallrye.beanbag.sisu.Sisu.addFieldInjections (Sisu.java:801)
    at io.smallrye.beanbag.sisu.Sisu.addClass (Sisu.java:580)
    at io.smallrye.beanbag.sisu.Sisu.addClassLoader (Sisu.java:91)
    at io.smallrye.beanbag.maven.MavenFactory.<init> (MavenFactory.java:71)
    at io.smallrye.beanbag.maven.MavenFactory.create (MavenFactory.java:96)
    at io.quarkus.maven.components.QuarkusWorkspaceProvider$1.configureMavenFactory (QuarkusWorkspaceProvider.java:88)
    at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.initRepoSystemAndManager (BootstrapMavenContext.java:876)
    at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.getRepositorySystem (BootstrapMavenContext.java:292)
    at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.<init> (MavenArtifactResolver.java:114)
    at io.quarkus.maven.components.QuarkusWorkspaceProvider.createArtifactResolver (QuarkusWorkspaceProvider.java:115)
    at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.artifactResolver (QuarkusBootstrapProvider.java:181)
    at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.doBootstrap (QuarkusBootstrapProvider.java:207)
    at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.bootstrapApplication (QuarkusBootstrapProvider.java:351)
    at io.quarkus.maven.QuarkusBootstrapProvider.bootstrapApplication (QuarkusBootstrapProvider.java:128)
    at io.quarkus.maven.QuarkusBootstrapMojo.bootstrapApplication (QuarkusBootstrapMojo.java:294)
    at io.quarkus.maven.GenerateCodeMojo.generateCode (GenerateCodeMojo.java:79)
    at io.quarkus.maven.GenerateCodeTestsMojo.doExecute (GenerateCodeTestsMojo.java:13)
    at io.quarkus.maven.QuarkusBootstrapMojo.execute (QuarkusBootstrapMojo.java:169)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
```

Same project is built fine with Maven 3.9.8, but mvnd 1.0.0 explodes as that above (but is based on very same Maven 3.9.8).

Reason: mvnd _includes_ plexus-interactivity-api and is provided by it, and it has **optional dependency on jline-reader**:
https://github.com/codehaus-plexus/plexus-interactivity/blob/master/plexus-interactivity-api/pom.xml#L17-L22

Beanbag, if want to "mimic" Sisu container, should _silently ignore this_ (the fact that class is not present).

In short: mvn 3.9.8 does not have plexus-interactivity-api present, while mvnd 1.0.0 does provide it, but without optional dependency (jline-reader). This makes beanbag choke on it.

--- a/sisu/src/main/java/io/smallrye/beanbag/sisu/Sisu.java
+++ b/sisu/src/main/java/io/smallrye/beanbag/sisu/Sisu.java
@@ -138,7 +138,11 @@ public final class Sisu {
             }
         }
         for (Component<?> component : map.values()) {
-            addBeanFromXml(component, filter, classLoader);
+            try {
+                addBeanFromXml(component, filter, classLoader);
+            } catch (LinkageError ignored) {
+                // skip
+            }
         }
     }
 
@@ -165,14 +169,12 @@ public final class Sisu {
                         if (className.isBlank() || builder.isTypeFilteredOut(className)) {
                             continue;
                         }
-                        final Class<?> clazz;
                         try {
-                            clazz = Class.forName(className, false, classLoader);
+                            final Class<?> clazz = Class.forName(className, false, classLoader);
+                            addClass(clazz, filter);
                         } catch (ClassNotFoundException | LinkageError ex) {
                             // todo: log it
-                            continue;
                         }
-                        addClass(clazz, filter);
                     }
                 }
             }
@@ -265,7 +267,11 @@ public final class Sisu {
                                 }
                                 if (new Annotations(clazz).getNamed() != null) {
                                     // it's a proper component; use the annotations to parse it
-                                    addClass(clazz, filter);
+                                    try {
+                                        addClass(clazz, filter);
+                                    } catch (LinkageError ex) {
+                                        // todo: log it
+                                    }
                                     consume(xr);
                                     return;
                                 }

